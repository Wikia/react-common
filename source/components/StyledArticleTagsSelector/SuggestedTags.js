import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

import StyledTag from '../StyledTag';

import TagShape from './TagShape';

const Tag = styled(StyledTag)`
    margin-right: 10px;
    margin-top: 10px;
`;

const Wrapper = styled.div`
    margin-top: 30px;
    padding-top: 8px;
`;

const SuggestedLabel = styled.div`
    color: ${({ theme }) => theme.color.fandom_dark_gray};
    font-size: ${({ theme }) => theme.font_size.xs};
`;

function SuggestedTags({
    className,
    tags,
    onClick,
}) {
    if (Array.isArray(tags) && tags.length > 0) {
        return (
            <Wrapper className={className}>
                <SuggestedLabel>Suggested</SuggestedLabel>
                {tags.map(({ id, title }) => (
                    <Tag
                        key={id}
                        onClick={() => onClick(id)}
                        tag={title}
                        secondary
                    />
                ))}
            </Wrapper>
        );
    }

    return null;
}

SuggestedTags.propTypes = {
    /** Extra class name */
    className: PropTypes.string,
    onClick: PropTypes.func,
    tags: PropTypes.arrayOf(TagShape),
};

SuggestedTags.defaultProps = {
    className: '',
    onClick: noop,
    tags: null,
};

export default SuggestedTags;
