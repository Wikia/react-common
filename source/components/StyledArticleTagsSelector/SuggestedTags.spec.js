import React from 'react';

import SuggestedTags from './SuggestedTags';

const tags = [
    { id: '1', title: 'Foo' },
    { id: '2', title: 'Bar' },
    { id: '3', title: 'FooBar' },
];

test('SuggestedTags renders with default props', () => {
    const component = mountWithThemeProvider(<SuggestedTags className="foo" tags={tags} />);
    expect(component.find(SuggestedTags)).toMatchSnapshot();
});

test('SuggestedTags does not render empty', () => {
    const component = mountWithThemeProvider(<SuggestedTags />);
    expect(component.text()).toBe('');
});
