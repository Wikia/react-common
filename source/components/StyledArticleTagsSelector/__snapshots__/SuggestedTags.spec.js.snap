// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedTags renders with default props 1`] = `
<SuggestedTags
  className="foo"
  onClick={[Function]}
  tags={
    Array [
      Object {
        "articleId": "1",
        "articleTitle": "Foo",
      },
      Object {
        "articleId": "2",
        "articleTitle": "Bar",
      },
      Object {
        "articleId": "3",
        "articleTitle": "FooBar",
      },
    ]
  }
>
  <SuggestedTags__Wrapper
    className="foo"
  >
    <div
      className="SuggestedTags__Wrapper-yazuu8-1 kDEEhr foo"
    >
      <SuggestedTags__SuggestedLabel>
        <div
          className="SuggestedTags__SuggestedLabel-yazuu8-2 iTNNRu"
        >
          Suggested
        </div>
      </SuggestedTags__SuggestedLabel>
      <SuggestedTags__Tag
        key="1"
        onClick={[Function]}
        secondary={true}
        tag="Foo"
      >
        <StyledTag
          className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
          onClick={[Function]}
          onRemove={null}
          removable={false}
          secondary={true}
          tag="Foo"
        >
          <StyledTag__Wrapper
            className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
            onClick={[Function]}
            secondary={true}
          >
            <span
              className="StyledTag__Wrapper-ja2wb4-0 iWktsc SuggestedTags__Tag-yazuu8-0 bOGtrE"
              onClick={[Function]}
            >
              Foo
            </span>
          </StyledTag__Wrapper>
        </StyledTag>
      </SuggestedTags__Tag>
      <SuggestedTags__Tag
        key="2"
        onClick={[Function]}
        secondary={true}
        tag="Bar"
      >
        <StyledTag
          className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
          onClick={[Function]}
          onRemove={null}
          removable={false}
          secondary={true}
          tag="Bar"
        >
          <StyledTag__Wrapper
            className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
            onClick={[Function]}
            secondary={true}
          >
            <span
              className="StyledTag__Wrapper-ja2wb4-0 iWktsc SuggestedTags__Tag-yazuu8-0 bOGtrE"
              onClick={[Function]}
            >
              Bar
            </span>
          </StyledTag__Wrapper>
        </StyledTag>
      </SuggestedTags__Tag>
      <SuggestedTags__Tag
        key="3"
        onClick={[Function]}
        secondary={true}
        tag="FooBar"
      >
        <StyledTag
          className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
          onClick={[Function]}
          onRemove={null}
          removable={false}
          secondary={true}
          tag="FooBar"
        >
          <StyledTag__Wrapper
            className="SuggestedTags__Tag-yazuu8-0 bOGtrE"
            onClick={[Function]}
            secondary={true}
          >
            <span
              className="StyledTag__Wrapper-ja2wb4-0 iWktsc SuggestedTags__Tag-yazuu8-0 bOGtrE"
              onClick={[Function]}
            >
              FooBar
            </span>
          </StyledTag__Wrapper>
        </StyledTag>
      </SuggestedTags__Tag>
    </div>
  </SuggestedTags__Wrapper>
</SuggestedTags>
`;
